pages_domain.go

```diff
@@ -0,0 +1,181 @@
+	// Make sure account id is set
+	if params.AccountID == "" {
+		return []PagesDomain{}, ErrMissingAccountID
+	}
+
+	// Make sure project name is set
+	if params.ProjectName == "" {
+		return []PagesDomain{}, ErrMissingProjectName
+	}
+
+	uri := fmt.Sprintf("/accounts/%s/pages/projects/%s/domains", params.AccountID, params.ProjectName)
+	res, err := api.makeRequestContext(ctx, http.MethodGet, uri, nil)
+	if err != nil {
+		return []PagesDomain{}, err
+	}
+
+	var pageDomainResponse PagesDomainsResponse
+	if err := json.Unmarshal(res, &pageDomainResponse); err != nil {
+		return []PagesDomain{}, err
+	}
+	return pageDomainResponse.Result, nil
+}
+
+// GetPagesDomain gets a single domain.
+//
+// API Reference: https://api.cloudflare.com/#pages-domains-get-domains
+func (api *API) GetPagesDomain(ctx context.Context, params PagesDomainParameters) (PagesDomain, error) {
+	// Make sure account id is set
+	if params.AccountID == "" {
+		return PagesDomain{}, ErrMissingAccountID
+	}
+
+	// Make sure project name is set
+	if params.ProjectName == "" {
+		return PagesDomain{}, ErrMissingProjectName
+	}
+	// Make sure domain name is set
+	if params.DomainName == "" {
+		return PagesDomain{}, ErrMissingDomain
+	}
+
+	uri := fmt.Sprintf("/accounts/%s/pages/projects/%s/domains/%s", params.AccountID, params.ProjectName, params.DomainName)
+	res, err := api.makeRequestContext(ctx, http.MethodGet, uri, nil)
+	if err != nil {
+		return PagesDomain{}, err
+	}
+
+	var pagesDomainResponse PagesDomainResponse
+	if err := json.Unmarshal(res, &pagesDomainResponse); err != nil {
+		return PagesDomain{}, err
+	}
+	return pagesDomainResponse.Result, nil
+}
+
+// PagesPatchDomain retries the validation status of a single domain.
+//
+// API Reference: https://api.cloudflare.com/#pages-domains-patch-domain
+func (api *API) PagesPatchDomain(ctx context.Context, params PagesDomainParameters) (PagesDomain, error) {
+	// Make sure account id is set
+	if params.AccountID == "" {
+		return PagesDomain{}, ErrMissingAccountID
+	}
+
+	// Make sure project name is set
+	if params.ProjectName == "" {
+		return PagesDomain{}, ErrMissingProjectName
+	}
+	// Make sure domain name is set
+	if params.DomainName == "" {
+		return PagesDomain{}, ErrMissingDomain
+	}
+
+	uri := fmt.Sprintf("/accounts/%s/pages/projects/%s/domains/%s", params.AccountID, params.ProjectName, params.DomainName)
+	res, err := api.makeRequestContext(ctx, http.MethodPatch, uri, nil)
+	if err != nil {
+		return PagesDomain{}, err
+	}
+
+	var pagesDomainResponse PagesDomainResponse
+	if err := json.Unmarshal(res, &pagesDomainResponse); err != nil {
+		return PagesDomain{}, err
+	}
+	return pagesDomainResponse.Result, nil
+}
+
+// PagesAddDomain adds a domain to a pages project.
+//
+// API Reference: https://community.cloudflare.com/t/add-pages-domain-via-api/303510/3
+// Blame Washly for not making it official.
```

Comment:
@WalshyDev any chance of this documentation landing soon? i'd rather we didn't use a forum post as the API reference here.

Reply:
Yeah I can PR it in today.

Reply:
legend! thank you!

Reply:
i'm tracking this internally so we're 👍 to merge this when we address the couple of feedback items.